packages/sdk/src/client/recovery/actions/oidc.ts

```diff
@@ -0,0 +1,152 @@
+import { ByteVector } from "zksync-sso-circuits";
+
+import { OidcRecoveryModuleAbi, WebAuthModuleAbi } from "../../../abi/index.js";
+import { noThrow } from "../../../utils/helpers.js";
+import { getEip712Domain } from "../../utils/getEip712Domain.js";
+
+export type OidcData = {
+  oidcDigest: Hex;
+  iss: Hex;
+  aud: Hex;
+};
+export type ParsedOidcData = {
+  oidcDigest: string;
+  iss: string;
+  aud: string;
+};
+export const parseOidcData = (oidcData: OidcData): ParsedOidcData => {
+  const hexToAscii = (hex: Hex): string => ByteVector.fromHex(hex).toAsciiStr();
+
+  return {
+    oidcDigest: oidcData.oidcDigest.toString(), // Do not convert. oidcDigest is just a hash
+    iss: hexToAscii(oidcData.iss),
+    aud: hexToAscii(oidcData.aud),
+  };
+};
+export type AddOidcAccountArgs = {
+  contracts: {
+    recoveryOidc: Address; // oidc recovery module
+  };
+  paymaster?: {
+    address: Address;
+    paymasterInput?: Hex;
+  };
+  oidcData: OidcData;
+  onTransactionSent?: (hash: Hash) => void;
+};
+export type AddOidcAccountReturnType = {
+  transactionReceipt: TransactionReceipt;
+};
+export const addOidcAccount = async <
+  transport extends Transport,
+  chain extends Chain,
+  account extends Account,
+>(client: Client<transport, chain, account>, args: Prettify<AddOidcAccountArgs>): Promise<Prettify<AddOidcAccountReturnType>> => {
+  const encodedOidcData = encodeAbiParameters(
+    [
+      {
+        type: "tuple", name: "OidcData", components: [
+          { type: "bytes32", name: "oidcDigest" },
+          { type: "bytes", name: "iss" },
+          { type: "bytes", name: "aud" },
+        ],
+      },
+    ],
+    [args.oidcData],
+  );
+
+  const callData = encodeFunctionData({
+    abi: OidcRecoveryModuleAbi,
+    functionName: "addValidationKey",
+    args: [encodedOidcData],
+  });
+
+  const sendTransactionArgs = {
+    account: client.account,
+    to: args.contracts.recoveryOidc,
+    paymaster: args.paymaster?.address,
+    paymasterInput: args.paymaster?.address ? (args.paymaster?.paymasterInput || getGeneralPaymasterInput({ innerInput: "0x" })) : undefined,
+    data: callData,
+    gas: 10_000_000n, // TODO: Remove when gas estimation is fixed
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } as any;
+
+  const transactionHash = await sendTransaction(client, sendTransactionArgs);
+  if (args.onTransactionSent) {
+    noThrow(() => args.onTransactionSent?.(transactionHash));
+  }
+
+  const transactionReceipt = await waitForTransactionReceipt(client, { hash: transactionHash });
+  if (transactionReceipt.status !== "success") throw new Error("addOidcAccount transaction reverted");
+
+  return {
+    transactionReceipt,
+  };
+};
+
+export type CalculateAddKeyHashArgs = {
+  passkeyPubKey: [Hex, Hex];
+  passkeyDomain: string;
+  contracts: {
+    recoveryOidc: Address; // oidc recovery module
+    passkey: Address;
+  };
+};
+
+export const calculateAddKeyTxHash = async <
```

Comment:
We need to calculate the hash of the tx that we are going to broadcast to set the new passkey. That's what this method method does.